Fix missing styles for dynamic routes rendering `<StarlightPage>`

HiDeoo opened the pull request:
#### Description

- Related to https://github.com/delucis/astro-embed/issues/185

This PR introduces a test and a workaround for an issue happening when a dyanmic route alphabetically sorted before Starlight route (`[...slug]`) renders a `<StarlightPage>` component. Just by having such setup, an unrelated page, e.g. the home page, rendering a component that includes some styles, will not have the styles applied due to an [head propagation issue](https://github.com/withastro/astro/issues/13724) (the linked issue contains more details about it).

This is a dev-only issue, as the production build works as expected.

This PR includes a workaround based on the investigation I made, detailed in https://github.com/withastro/astro/issues/13724 by adding an extra import to `astro:content` which are key to head propagation and an E2E test to ensure we don't regress on it in the future.

Depending on how easy/doable it is to fix the issue in Astro, we can either decide to ship this workaround or not.

astrobot-houston commented:
<!-- lunaria-action-comment -->

## Lunaria Status Overview


🌕 **This pull request will trigger status changes.**

<details>
<summary>Learn more</summary>

By default, every PR changing files present in the [Lunaria configuration's `files` property](https://lunaria.dev/reference/configuration/#files-required) will be considered and trigger status changes accordingly.

You can change this by adding one of the keywords present in the [`ignoreKeywords` property in your Lunaria configuration file](https://lunaria.dev/reference/configuration/#ignorekeywords) in the PR's title (ignoring all files) or by [including a tracker directive](https://lunaria.dev/guides/tracking/#tracker-directives) in the merged commit's description. 
</details>

### Tracked Files




| Locale | File                                                                                                                                      | Note                                                       |
| ------ | ----------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------- |
| en     | [index.mdx](https://github.com/withastro/starlight/blob/65283708faebaa201c366b3bfee20a4fbeb2b264/docs%2Fsrc%2Fcontent%2Fdocs%2Findex.mdx) | Source changed, localizations will be marked as outdated.  |

<details>
<summary>Warnings reference</summary>

| Icon | Description                                                                                                                                       |
| ---- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
| 🔄️  | The source for this localization has been updated since the creation of this pull request, make sure all changes in the source have been applied. |

</details>

HiDeoo commented:
I've updated the PR and its description with everything I learned from my investigation. I also refactored the workaround as I now understand the issue better and included some comments and an E2E test to ensure we don't break it in the future.

I've also tested the workaround locally for the Biome website using https://github.com/biomejs/website/pull/2082

| Before   | After    |
| -------- | -------- |
| <img width="1115" alt="SCR-20250430-jupd" src="https://github.com/user-attachments/assets/1f324317-7f85-4d63-bd9f-6c8781421007" /> | <img width="1097" alt="SCR-20250430-juhw" src="https://github.com/user-attachments/assets/b1cbc64d-b902-4198-b0b6-96205121a00b" /> |

delucis reviewed (approved):
Just noting that this all looks good to me if you decide we should ship this as a workaround for the bigger Astro issue. Thanks for the in-depth research, testing, and great code comments!

HiDeoo commented on packages/starlight/__e2e__/components.test.ts:
```diff
@@ -393,6 +393,26 @@ test.describe('anchor headings', () => {
+		page,
+		makeServer,
+	}) => {
+		const starlight = await makeServer('dev', { mode: 'dev' });
```
This is actually a dev only issue. I've mentioned it in the 2 large comments I added but I totally missed adding it in the test name. I'll update the test name too so it's more obvious it's dev-only 👍

HiDeoo commented:
Updated the PR with the latest comments and also added a changeset. Not quite sure the level of detail we should include in the changeset, but other than that, we should be good to go.

delucis reviewed (approved):
Thanks again @HiDeoo! I think the changeset is OK — the full detail of the when/why is a bit complicated and the main thing is covered: if styles were missing, this release might fix it.
